src/transform/src/map_lifting.rs

```diff
@@ -494,10 +494,11 @@ impl LiteralLifting {
 
                 let eval_constant_aggr = |aggr: &expr::AggregateExpr| {
                     let temp = repr::RowArena::new();
-                    let eval = aggr
-                        .func
-                        .eval(Some(aggr.expr.eval(&[], &temp).unwrap()), &temp);
-                    MirScalarExpr::literal_ok(
+                    let mut eval = aggr.expr.eval(&[], &temp);
```

Comment:
This was the original suggested fix, but this code is not exercised anymore. I guess it doesn't hurt, but it would be good to find a way of exercising it.

Reply:
Idk if it fights with column_knowledge, but probably
```sql
select sum(1/x) from (select y, 0 as x from table)
```
might do it? The `table` prevents constant folding from applying, and either `map_lifting` or `column_knowledge` gets to do the work.

Reply:
Oh, good one. Although `sum` doesn't make it crash because we don't consider it as a constant expression when the parameter is a literal, since it still depends on the number of input rows. However, `min` does the job and makes the system crash without this fix:

```sql
select min(1/x) from (select a as y, 0 as x from a);
```

I wonder if we should treat any aggregate function with an error parameter as a constant expression. Diff would be:

```
diff --git a/src/expr/src/relation/mod.rs b/src/expr/src/relation/mod.rs
index f2f0478a7..8ff1810e0 100644
--- a/src/expr/src/relation/mod.rs
+++ b/src/expr/src/relation/mod.rs
@@ -1400,7 +1400,7 @@ impl AggregateExpr {
             | AggregateFunc::MinTimestampTz
             | AggregateFunc::Any
             | AggregateFunc::All => self.expr.is_literal(),
-            _ => false,
+            _ => self.expr.is_literal_err(),
         }
     }
 }
```

Any thoughts on that?

Reply:
I'd be in favor of this change since if we have multiple rows in a group, we don't need multiple copies of the error or to sum the error into a super-error.

Reply:
Agreed. I've added this change.